### PR TITLE
[kortex_api] use CMake FetchContent to get API

### PR DESCRIPTION
**Pull request by moriarty**
- This uses CMake FetchContent to get the API.
- Currently only Linux x86_64 is supported
- Adds comment for future use of ExternalProject_Add regardind detecting and pulling other versions of the API

### CODE REVIEW

**Comment by abake48** [comment]
With this PR, could we remove the `Kortex_api` package from this repo?

**Comment by moriarty** [comment]
> With this PR, could we remove the `Kortex_api` package from this repo?

Not the kortex_api package, but all the files except for CMakeLists.txt and package.xml
I didn't do it all at once because we always squash merge. Locally I have been testing all the files and I wanted to trigger the full CI

File: kortex_api/CMakeLists.txt [hunk 1]
```diff
@@ -1,17 +1,49 @@
...
+#)
+
 find_package(ament_cmake REQUIRED)
 
-ament_export_libraries(${CMAKE_CURRENT_SOURCE_DIR}/lib/release/libKortexApiCpp.a)
+ament_export_libraries(${kinova_binary_api_SOURCE_DIR}/lib/release/libKortexApiCpp.a)
 link_libraries(pthread)
 
-ament_export_include_directories(include)
+ament_export_include_directories(${kinova_binary_api_SOURCE_DIR}/include)
```
Comment (moriarty) [inline kortex_api/CMakeLists.txt]:
```suggestion
ament_export_include_directories(include)
```

**Review by abake48 — APPROVED** [review]
Tested by building this branch in the Studio dev container on the Debian machine connected to our Gen 3 arm.

**Pull request merged**